src/Compilers/VisualBasic/Portable/Emit/EditAndContinue/VisualBasicSymbolMatcher.vb

```diff
@@ -313,13 +297,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Emit
             End Function
 
             Public Overrides Function VisitModule([module] As ModuleSymbol) As Symbol
-                ' Only map symbols from source assembly and its previous generations to the other assembly. 
-                ' All other symbols should map to themselves.
-                If [module].ContainingAssembly.Identity.Equals(_sourceAssembly.Identity) Then
-                    Return _otherAssembly.Modules([module].Ordinal)
-                Else
-                    Return [module]
+                Dim otherAssembly = DirectCast(Visit([module].ContainingAssembly), AssemblySymbol)
+                If otherAssembly Is Nothing Then
+                    Return Nothing
+                End If
+
+                If [module].Ordinal = 0 Then
+                    Return otherAssembly.Modules(0)
                 End If
+
+                For i = 1 To otherAssembly.Modules.Length - 1
+                    Dim otherModule = otherAssembly.Modules(i)
+                    If otherModule.Name = [module].Name Then
```

Comment:
String comparison in VB can be affected by Option Compare, consider calling comparison method explicitly.